app/assets/javascripts/controllers/generic_object/generic_object_definition_controller.js

```diff
@@ -0,0 +1,75 @@
+ManageIQ.angular.app.controller('genericObjectDefinitionFormController', ['$http', '$scope', 'miqService', 'genericObjectSubscriptionService', function($http, $scope, miqService, genericObjectSubscriptionService) {
+  var init = function() {
+    hideAndClearForm();
+
+    $scope.newRecord = true;
+    $scope.showSingleItem = false;
+
+    genericObjectSubscriptionService.subscribeToShowAddForm(showAddForm);
+    genericObjectSubscriptionService.subscribeToTreeClicks(showSelectedItem);
+    genericObjectSubscriptionService.subscribeToRootTreeclicks(showAllItems);
+  };
+
+  var showAddForm = function(_response) {
+    $scope.clearForm();
+    $scope.showAddForm = true;
+    sendDataWithRx({eventType: 'deselectTreeNodes'});
+  };
+
+  var showAllItems = function(response) {
+    $scope.genericObjectList = response;
+    $scope.showSingleItem = false;
+  };
+
+  var addedGenericObject = function(data) {
```

Comment:
It's using promises, but not angular's `$q` promises .. and we're kinda skipping the promise at all by passing it in `.done`, not calling `.then` on the result.

You *could* fix it by doing `.then` instead, and then fixing `miqService.jQueryRequest` to a) actually return the value, and b) wrap it in `$q.when(...)` so that it becomes an angulary-promise that knows about digest..